src/groupByKey.ts

```diff
@@ -2,6 +2,33 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Value = string | number;
+
+type Obj = {
+  [key: string]: Value;
+};
+
+export function groupByKey(items: Obj[], key: keyof Obj): GroupsMap<Obj> {
+  const result: GroupsMap<Obj> = {};
+  const moddedItems = items.map((item1) => {
+    const arrayForItems: Obj[] = [];
+
+    items.filter((item2) => {
+      if (item2[key] === item1[key]) {
+        arrayForItems.push(item2);
+
+        return true;
+      }
+
+      return false;
+    });
+
+    return arrayForItems;
+  });
```

Comment:
The `map` method is not suitable here because it creates a new array with the same length as the original array. But you need to group items by key, so the length of the resulting object can be less than the length of the original array. Also, the `filter` method inside the `map` method creates a new array for each item in the original array. This is inefficient and unnecessary. You can solve the task using one loop and without creating additional arrays. Use the `forEach` method instead of the `map` method.